java/src/main/java/ai/rapids/cudf/MemoryCleaner.java

```diff
@@ -197,30 +198,46 @@ static void setDefaultGpu(int defaultGpuId) {
     }
   }, "Cleaner Thread");
 
+  /**
+   * Runnable used to be added to Java default shutdown hook.
+   * It checks the leaks at shutdown time.
+   * If you want to register the Runnable in a custom shutdown hook manager instead of Java default shutdown hook,
+   * should first remove it and then add it.
+   */
+  public static final Runnable DEFAULT_SHUTDOWN_RUNNABLE = () -> {
+    // If we are debugging things do a best effort to check for leaks at the end
+
+    System.gc();
+    // Avoid issues on shutdown with the cleaner thread.
+    t.interrupt();
+    try {
+      t.join(1000);
+    } catch (InterruptedException e) {
+      // Ignored
+    }
+    if (defaultGpu >= 0) {
+      Cuda.setDevice(defaultGpu);
+    }
+
+    for (CleanerWeakReference cwr : all) {
+      cwr.clean();
+    }
+  };
+
+  private static final Thread DEFAULT_SHUTDOWN_THREAD = new Thread(DEFAULT_SHUTDOWN_RUNNABLE);
+
   static {
     t.setDaemon(true);
     t.start();
     if (REF_COUNT_DEBUG) {
-      // If we are debugging things do a best effort to check for leaks at the end
-      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-        System.gc();
-        // Avoid issues on shutdown with the cleaner thread.
-        t.interrupt();
-        try {
-          t.join(1000);
-        } catch (InterruptedException e) {
-          // Ignored
-        }
-        if (defaultGpu >= 0) {
-          Cuda.setDevice(defaultGpu);
-        }
-        for (CleanerWeakReference cwr : all) {
-          cwr.clean();
-        }
-      }));
+      Runtime.getRuntime().addShutdownHook(DEFAULT_SHUTDOWN_THREAD);
     }
   }
 
+  public static void removeDefaultShutdownHook() {
```

Comment:
javadoc for the public method?

to reduce the risk of race conditions we could change the signature of this method 
```java
public static Runnable removeDefaultShutdownHook() 
```
after removing `DEFAULT_SHUTDOWN_THREAD` it would return `DEFAULT_SHUTDOWN_RUNNABLE` that the caller can register in their shutdown manager

DEFAULT_SHUTDOWN_RUNNABLE can be `private` then

Reply:
Done